src/firebase_functions/scheduler_fn.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Cloud functions to handle Schedule triggers."""
+
+import typing as _typing
+import dataclasses as _dataclasses
+import datetime as _dt
+import functools as _functools
+
+import firebase_functions.options as _options
+import firebase_functions.private.util as _util
+from functions_framework import logging as _logging
+from flask import (
+    Request as _Request,
+    Response as _Response,
+    make_response as _make_response,
+)
+
+# Export for user convenience.
+# pylint: disable=unused-import
+from firebase_functions.options import Timezone
+
+
+@_dataclasses.dataclass(frozen=True)
+class ScheduledEvent:
+    """
+    A ScheduleEvent that is passed to the function handler
+    """
+
+    job_name: str | None
+    """
+    The Cloud Scheduler job name.
+    Populated via the X-CloudScheduler-JobName header.
+    If invoked manually, this field is None.
+    """
+
+    schedule_time: _dt.datetime
```

Comment:
nit* This is great! I'd been fine with string, but parsing it sounds extra helpful!